playkit/src/main/java/com/kaltura/playkit/PKMediaEntry.java

```diff
@@ -86,13 +88,21 @@ public int describeContents() {
     public void writeToParcel(Parcel dest, int flags) {
         dest.writeString(this.id);
         dest.writeString(this.name);
-        dest.writeTypedList(this.sources);
+        if (sources != null) {
+            dest.writeTypedList(this.sources);
+        } else {
+            dest.writeTypedList(Collections.EMPTY_LIST);
+        }
         dest.writeLong(this.duration);
         dest.writeInt(this.mediaType == null ? -1 : this.mediaType.ordinal());
-        dest.writeInt(this.metadata.size());
-        for (Map.Entry<String, String> entry : this.metadata.entrySet()) {
-            dest.writeString(entry.getKey());
-            dest.writeString(entry.getValue());
+        if (this.metadata != null) {
+            dest.writeInt(this.metadata.size());
+            for (Map.Entry<String, String> entry : this.metadata.entrySet()) {
+                dest.writeString(entry.getKey());
+                dest.writeString(entry.getValue());
+            }
+        } else {
+            dest.writeBundle(new Bundle());
```

Comment:
if you're using the for..write key/value method to write the map to the parcel you need to use the same 
way to handle null metadata case. you can't use bundle when its null and key/value when it's not.
so if it's null writeInt(-1) to indicate no map content available and on the read from the parcel leave
this field null, or - 0 - if you want to create an empty list.
so please remove the writeBundle and use one of the options mentioned above.